cobalt/browser/h5vcc_experiments/h5vcc_experiments_impl.cc

```diff
@@ -0,0 +1,117 @@
+  base::FeatureList::OverrideState override =
+      accessor->GetOverrideStateByFeatureName(feature_name);
+  if (override == base::FeatureList::OverrideState::OVERRIDE_ENABLE_FEATURE) {
+    return h5vcc_experiments::mojom::OverrideState::OVERRIDE_ENABLE_FEATURE;
+  } else if (override ==
+             base::FeatureList::OverrideState::OVERRIDE_DISABLE_FEATURE) {
+    return h5vcc_experiments::mojom::OverrideState::OVERRIDE_DISABLE_FEATURE;
+  }
+  return h5vcc_experiments::mojom::OverrideState::OVERRIDE_USE_DEFAULT;
+}
+
+std::string GetFeatureParamInternal(const std::string& feature_param_name) {
+  return base::GetFieldTrialParamValue(cobalt::kCobaltExperimentName,
+                                       feature_param_name);
+}
+
+}  // namespace
+
+H5vccExperimentsImpl::H5vccExperimentsImpl(
+    content::RenderFrameHost& render_frame_host,
+    mojo::PendingReceiver<mojom::H5vccExperiments> receiver)
+    : content::DocumentService<mojom::H5vccExperiments>(render_frame_host,
+                                                        std::move(receiver)) {}
+
+void H5vccExperimentsImpl::Create(
+    content::RenderFrameHost* render_frame_host,
+    mojo::PendingReceiver<mojom::H5vccExperiments> receiver) {
+  new H5vccExperimentsImpl(*render_frame_host, std::move(receiver));
+}
+
+void H5vccExperimentsImpl::SetExperimentState(
+    base::Value::Dict experiment_config,
+    SetExperimentStateCallback callback) {
+  auto experiment_config_ptr =
+      cobalt::GlobalFeatures::GetInstance()->experiment_config();
+
+  experiment_config_ptr->SetDict(
+      cobalt::kExperimentConfigFeatures,
+      std::move(experiment_config.Find(cobalt::kExperimentConfigFeatures)
+                    ->GetDict()));
+  experiment_config_ptr->SetDict(
+      cobalt::kExperimentConfigFeatureParams,
+      std::move(experiment_config.Find(cobalt::kExperimentConfigFeatureParams)
+                    ->GetDict()));
+  experiment_config_ptr->SetList(
+      cobalt::kExperimentConfigExpIds,
+      std::move(
+          experiment_config.Find(cobalt::kExperimentConfigExpIds)->GetList()));
+  std::move(callback).Run();
+}
+
+void H5vccExperimentsImpl::ResetExperimentState(
+    ResetExperimentStateCallback callback) {
+  PrefService* experiment_config =
+      cobalt::GlobalFeatures::GetInstance()->experiment_config();
+  experiment_config->ClearPref(cobalt::kExperimentConfigFeatures);
+  experiment_config->ClearPref(cobalt::kExperimentConfigFeatureParams);
+  experiment_config->ClearPref(cobalt::kExperimentConfigExpIds);
+  std::move(callback).Run();
+}
+
+void H5vccExperimentsImpl::GetActiveExperimentIds(
+    GetActiveExperimentIdsCallback callback) {
+  std::vector<uint32_t> active_experiment_ids;
+  for (auto& experiment_id :
+       cobalt::GlobalFeatures::GetInstance()->experiment_config()->GetList(
+           cobalt::kExperimentConfigExpIds)) {
+    active_experiment_ids.push_back(experiment_id.GetInt());
```

Comment:
nit: This will grow `active_experiment_ids` (with all the fancy std::vector
grow stuff), instead we can pre-reserve its capacity by e.g.
``` 
  const auto experiments = cobalt::GlobalFeatures::GetInstance()->
                             experiment_config()->GetList(
                               cobalt::kExperimentConfigExpIds))
  std::vector<uint32_t> active_experiment_ids;
  active_experiments_id.reserve(experiments.size());
```
 etc